### PR TITLE
[cli] update dataset to operate directly on TLVs

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/cli/cli_dataset.cpp [hunk 1]
```diff
@@ -336,19 +344,23 @@ template <> otError Dataset::Process<Cmd("channel")>(Arg aArgs[])
...
     }
     else
     {
-        SuccessOrExit(error = aArgs[0].ParseAsUint32(sDataset.mChannelMask));
-        sDataset.mComponents.mIsChannelMaskPresent = true;
+        memset(&dataset, 0, sizeof(dataset));
+        SuccessOrExit(error = aArgs[0].ParseAsUint32(dataset.mChannelMask));
+        dataset.mComponents.mIsChannelMaskPresent = true;
+        SuccessOrExit(error = otDatasetUpdateTlvs(&dataset, &sDataset));
     }
```
Comment (abtink) [inline src/cli/cli_dataset.cpp:364]: In the `else` part we need to clear all `mIs{}Present`?
Maybe we can add a helper method here `Clear(otOperationalDataset &aDataset)` and do `memset(0)`?

Comment (abtink) [reply src/cli/cli_dataset.cpp:364]: I think you already did it 😄

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

**Review by wgtdkp — APPROVED** [review]
Thanks! :+1:

**Pull request merged**